compiler/sem/pragmas.nim

```diff
@@ -1438,119 +1299,53 @@ proc prepareSinglePragma(
           incl(sym.flags, sfError)
           excl(sym.flags, sfForward)
         else:
-          let s = getStrLitNode(c, it)
-          case s.kind:
-            of nkError:
-              result = s # err
-            else:
-              recordPragma(c, it, "error", s.strVal)
-              result = c.config.newError(
-                it, PAstDiag(kind: adSemCustomUserError, errmsg: s.strVal))
-      of wFatal:
-        result = c.config.newError(it, PAstDiag(kind: adSemFatalError))
-      of wDefine:
-        result = processDefine(c, it)
-      of wUndef:
-        result = processUndef(c, it)
-      of wCompile:
-        result = processCompile(c, it)
-      of wLink:
-        result = processLink(c, it)
-        result = it
-      of wPassl:
-        let (s, err) = strLitToStrOrErr(c, it)
-        result =
-          if err.isNil:
-            extccomp.addLinkOption(c.config, s)
-            recordPragma(c, it, "passl", s)
-            it
-          else:
-            err
-      of wPassc:
-        let (s, err) = strLitToStrOrErr(c, it)
-        result =
-          if err.isNil:
-            extccomp.addCompileOption(c.config, s)
-            recordPragma(c, it, "passc", s)
-            it
-          else:
-            err
-      of wLocalPassc:
-        assert sym != nil and sym.kind == skModule
-        let (s, err) = strLitToStrOrErr(c, it)
-        result =
-          if err.isNil:
-            extccomp.addLocalCompileOption(
-              c.config, s, toFullPathConsiderDirty(c.config, sym.info.fileIndex))
-            recordPragma(c, it, "localpassl", s)
-            it
-          else:
-            err
-      of wPush:
-        result = processPush(c, n, i + 1)
-        result.flags.incl nfImplicitPragma # xxx: legacy singlepragma=true
-      of wPop:
-        result = processPop(c, it)
-        result.flags.incl nfImplicitPragma # xxx: legacy singlepragma=true
+          # XXX: disallow for the other declarations (e.g. ``var``)?
```

Comment:
yeah, that's weird, because `wError` is in `declPragmas` whcih is under `varPragmas`. 🤔 I bet the rest of the compiler can't handle it.

Reply:
I restored the previous behaviour (`.error`s on, for example, a `var` being treated like they are in statements) for now.